src/itest/resources/features/company-profile-error-retry.feature

```diff
@@ -0,0 +1,48 @@
+Feature: Error and retry scenarios for company profile
+
+  Scenario Outline: Processing company profile information unsuccessfully
+
+    Given Company profile api service is running
+    When I send PATCH request with raw payload "<data>" and company number "<company_number>"
+    Then I should receive <response_code> status code
+    And the CHS Kafka API is not invoked
+    And nothing is persisted in the database
+
+    Examples:
+      | data                               | company_number     | response_code |
+      | bad_request_payload                | 11748564           | 400           |
+
+  Scenario Outline: Retrieve company profile unsuccessfully as the company number does not exist
+
+    Given Company profile api service is running
+    When I send GET request with company number "<data>"
+    Then I should receive 404 status code
+
+    Examples:
+      | data     |
+      | 00006400 |
+
```

Comment:
Scenario 3 is missing.

Scenario 3 (500 - Internal Server Error):
Given Company Profile API component is successfully running 
When ‘PATCH’ Rest endpoint is invoked with a valid json payload that causes a NPE (make some attribute in payload as null) NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 500 (Internal Error) to the client and data is not updated into Mongo DB, and CHS Kafka API is never invoked

Reply:
Changes done